samples/dotnet/kernel-syntax-examples/Example12_Planning.cs

```diff
@@ -23,6 +23,46 @@ public static async Task RunAsync()
         await EmailSamplesAsync();
         await BookSamplesAsync();
         await MemorySampleAsync();
+        await ConditionalSampleAsync();
+    }
+
+    private static async Task ConditionalSampleAsync()
+    {
+        Console.WriteLine("======== Planning - Conditional flow example ========");
+        var kernel = InitializeKernelAndPlanner(out var planner);
+
+        // Load additional skills to enable planner to do non-trivial asks.
+        string folder = RepoFiles.SampleSkillsPath();
+        kernel.ImportSemanticSkillFromDirectory(folder, "SummarizeSkill");
+        kernel.ImportSemanticSkillFromDirectory(folder, "WriterSkill");
+
+        var originalPlan = await kernel.RunAsync(
+            @"Summarize an outline about AI If the summarization contains any reference to ""AGI"" then Generate a summary outline for AGI.
+After that generate a list of synopsis for a novel using the resulting summarization.
+Generate a poem about love.", planner["CreatePlan"]);
+        // <goal>Summarize an outline about AI If the summarization length is greater than 100 then Generate a summary outline for AGI. After that generate a list of synopsis for a novel using the resulting summarization. Generate a poem about love</goal>
+        // <plan>
+        //   <function.Everything.Summarize setContextVariable=""SUMMARIZED_OUTLINE""/>
+        //   <if>
+        //     <conditiongroup>
+        //       <condition variable=""$SUMMARIZED_OUTLINE"" greaterthan=""100""/>
+        //     </conditiongroup>
+        //     <then>
+        //       <function.Everything.Summarize input=""Artificial General Intelligence (AGI) is the hypothetical ability of a machine to perform any intellectual task that a human can. It is a major goal of AI research and a common topic of science fiction and futurism. Some of the challenges and risks of developing AGI include ethical, social, philosophical, and existential issues."" setContextVariable=""SUMMARIZED_OUTLINE_AGI""/>
+        //       <function.Everything.GenerateList input=""$SUMMARIZED_OUTLINE_AGI"" listType=""synopsis"" listCount=""3"" setContextVariable=""SYNOPSIS_LIST""/>
+        //       <function.Everything.GeneratePoem input=""love"" poemType=""small""/>
+        //     </then>
+        //     <else>
+        //       <function.Everything.GenerateList input=""$SUMMARIZED_OUTLINE"" listType=""synopsis"" listCount=""3"" setContextVariable=""SYNOPSIS_LIST""/>
+        //       <function.Everything.GeneratePoem input=""love"" poemType=""small""/>
+        //     </else>
+        //   </if>
+        // </plan>
+
+        Console.WriteLine("Original plan:");
+        Console.WriteLine(originalPlan.Variables.ToPlan().PlanString);
+
+        await ExecutePlanAsync(kernel, planner, originalPlan, 5);
```

Comment:
Is 5 the required steps here for completion of the plan? What is the input to the first `Summarize`? Can you show it with two different inputs to show it goes down each branch?

Reply:
The number of steps can be greater, currently the plan will resolve the if (then/else) inner steps and add them as part of the list, so if in the overall branching we end up with more than 5 steps this will abort at the given step. For branching examples I can move this number up a bit.